search/collector/eligible.go

```diff
@@ -23,28 +23,40 @@ import (
 	index "github.com/blevesearch/bleve_index_api"
 )
 
+type eligibleStore struct {
+	ids []index.IndexInternalID
+}
+
+func (s *eligibleStore) AddDocumentMatch(doc *search.DocumentMatch) *search.DocumentMatch {
+	copyOfID := make([]byte, len(doc.IndexInternalID))
+	copy(copyOfID, doc.IndexInternalID)
+	s.ids = append(s.ids, copyOfID)
+	return doc
+}
+
 type EligibleCollector struct {
 	size    int
 	total   uint64
 	took    time.Duration
 	results search.DocumentMatchCollection
 
-	store collectorStore
+	store *eligibleStore
```

Comment:
This need not be a pointer looking at this contents.

Reply:
Also, this feels over done with the type routing, simply rename to `ids` (in-line with the method you're using to obtain this list) and make it of type `[]index.IndexInternalID`